### PR TITLE
Fix some issues with FulgoraGraphComputer

### PR DESCRIPTION
**Pull request by mad**
Add support for job without combiner - just combine message into list
Detach elements before it will be written into graph for avoid serialization issue

Now ConnectedComponentTest and ShortestPathVertexProgramTest passed

TP tests for `BerkeleyJanusGraphComputerTest` passed

Only two failed but on master too
```
WhereTest$Traversals
WhereTest$Traversals.g_V_hasXageX_asXaX_out_in_hasXageX_asXbX_selectXa_bX_whereXa_outXknowsX_bX
WhereTest$Traversals.g_V_asXaX_outXcreatedX_whereXasXaX_name_isXjoshXX_inXcreatedX_name
```
Possible cause is `PathProcessorStrategy` it generate random alias for step for each vertexJob iteration. If I setup [is.testing](https://github.com/apache/tinkerpop/blob/db363b53d8d6ea5d91c9a1f8f8159b8b8511d8c4/gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/strategy/optimization/PathProcessorStrategy.java#L69) that tests passed

Signed-off-by: Pavel Ershov <owner.mad.epa@gmail.com>

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there an issue associated with this PR? Is it referenced in the commit message?
- [ ] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [ ] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you written and/or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](https://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [ ] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?
- [ ] If this PR is a documentation-only change, have you added a `[doc only]`
  tag to the first line of your commit message to avoid spending CPU cycles in
  Travis CI when no code, tests, or build configuration are modified?

### Note:
Please ensure that once the PR is submitted, you check Travis CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by FlorianHockmann** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity decreasing per file
==============================
+ janusgraph-core/src/main/java/org/janusgraph/graphdb/olap/computer/VertexProgramScanJob.java  -3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/JanusGraph/janusgraph/pullRequest?prid=5110603&bid=16805046)

File: janusgraph-core/src/main/java/org/janusgraph/graphdb/olap/computer/FulgoraGraphComputer.java [hunk 1]
```diff
@@ -373,13 +378,21 @@ private Graph writeMutatedPropertiesBackIntoGraph() {
...
+            HaltedTraverserStrategy haltedTraverserStrategy = HaltedTraverserStrategy.reference();
             Map<Long, Map<String, Object>> mutatedProperties = Maps.transformValues(vertexMemory.getMutableVertexProperties(),
-                new Function<Map<String, Object>, Map<String, Object>>() {
-                    @Nullable
-                    @Override
-                    public Map<String, Object> apply(final Map<String, Object> o) {
-                        return Maps.filterKeys(o, s -> !VertexProgramHelper.isTransientVertexComputeKey(s, vertexProgram.getVertexComputeKeys()));
-                    }
+                (Function<Map<String, Object>, Map<String, Object>>) o -> {
+                    Map<String, Object> nonTransientKeys =
```
Comment (FlorianHockmann) [inline janusgraph-core/src/main/java/org/janusgraph/graphdb/olap/computer/FulgoraGraphComputer.java]: (nitpick) I tried to understand what this function does but failed. If you can move some of this into its own method and give that a good name that tells what the code does, then that would be helpful to any future reader in my opinion.
However, all the Fulgora code is written this way so this added code fits in nicely here 😅

File: janusgraph-core/src/main/java/org/janusgraph/core/JanusGraph.java [hunk 2]
```diff
@@ -30,7 +30,6 @@
 @Graph.OptIn(Graph.OptIn.SUITE_STRUCTURE_STANDARD)
 @Graph.OptIn(Graph.OptIn.SUITE_PROCESS_STANDARD)
 @Graph.OptIn(Graph.OptIn.SUITE_PROCESS_COMPUTER)
-@Graph.OptIn("org.janusgraph.blueprints.process.traversal.strategy.JanusGraphStrategySuite")
```
[the remark this answers is not in the document]
Comment (mad) [reply janusgraph-core/src/main/java/org/janusgraph/core/JanusGraph.java]: This file does not exists in the repo

[the remark this answers is not in the document]
Comment (FlorianHockmann) [reply janusgraph-core/src/main/java/org/janusgraph/core/JanusGraph.java]: Strange, but ok, that's of course a good reason to remove this.

**Comment by farodin91** [comment]
@FlorianHockmann @mad What is the state of this PR?

**Pull request merged**